src/include/migraphx/op/scatter.hpp

```diff
@@ -38,30 +46,41 @@ struct scatter
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
         check_shapes{inputs, *this}.has(3).standard();
-        return inputs.front();
+        // If non-packed, this converts to a packed output while preserving permutation of tensor
+        return inputs.front().with_lens(inputs.front().lens());
     }
 
     argument compute(const shape& output_shape, std::vector<argument> args) const
     {
         argument result{output_shape};
+        auto& self = static_cast<const Derived&>(*this);
+
         // max dimension in axis
         auto axis_dim_size = output_shape.lens()[axis];
+        // iterate through all elements in output
```

Comment:
This comment seems out of place. `visit_all` doesnt iterate through the elements.